File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0    
    count_list = []
    
    for i in range(left, right+1):
        count = 0
        for j in range(1, i+1):
            if i%j == 0:
                count += 1
        answer = answer+i if count%2 == 0 else answer-i
            
    return answer

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
import java.lang.Math;

class Solution {
    public int solution(int left, int right) {
        int answer = 0;
        int cnt;
        
        for(int j=left; j<=right; j++) {
            cnt = 0;
            for(int i=1; i<=Math.sqrt(j); i++) {
                if(j%i == 0) 
                    cnt = i*i == j ? cnt+1 : cnt+2;
            }
            answer = cnt%2 == 0 ? answer+j : answer-j;
        }
        return answer;
    }
}